src/V3EmitCFunc.cpp

```diff
@@ -765,9 +768,20 @@ string EmitCFunc::emitVarResetRecurse(const AstVar* varp, bool constructing,
                     out += cvtToStr(constp->num().edataWord(w)) + "U;\n";
                 }
             } else {
-                out += zeroit ? (slow ? "VL_ZERO_RESET_W(" : "VL_ZERO_W(") : "VL_RAND_RESET_W(";
+                out += zeroit ? (slow ? "VL_ZERO_RESET_W(" : "VL_ZERO_W(")
+                              : "VL_SCOPED_RAND_RESET_W(";
                 out += cvtToStr(dtypep->widthMin());
-                out += ", " + varNameProtected + suffix + ");\n";
+                out += ", " + varNameProtected + suffix;
+                if (!zeroit) {
+                    emitVarResetScopeHash();
+                    uint64_t salt = std::hash<std::string>{}(varp->prettyName());
```

Comment:
```suggestion
                    const uint64_t salt = std::hash<std::string>{}(varp->prettyName());
```

Reply:
C++ "Hash functions are only required to produce the same result for the same input within a single execution of a program".  So use VString functions instead, making a new one if needed.

Reply:
That's a bummer.

I dropped a hash function in `V3String` and also the runtime library.  Is there something better to do here?  I don't see any hash functions already associated with `VString`.  There is a SHA256 in there, but it speaks about 64B blocks and I don't think that's what we want for this application.  There's also `V3Hash` but that is intrinsically 32b.  Also, is there a reasonable way to share code between the compiler and the runtime?  I'm not seeing any examples.